// ng-bootstrap-spinner/esm5/lib/ng-spinner.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var NgSpinnerComponent = /** @class */ (function () {
    function NgSpinnerComponent() {
    }
    /**
     * @return {?}
     */
    NgSpinnerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    NgSpinnerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ng-spinner',
                    template: "<div [ngClass]=\"['text-' + this.color,'spinner-' + this.type]\" [ngStyle]=\"{'width': size+'rem','height':size+'rem'}\"\r\n  role=\"status\">\r\n  <span class=\"sr-only\">Loading...</span>\r\n</div>\r\n",
                    styles: [""]
                }] }
    ];
    /** @nocollapse */
    NgSpinnerComponent.ctorParameters = function () { return []; };
    NgSpinnerComponent.propDecorators = {
        type: [{ type: Input }],
        size: [{ type: Input }],
        color: [{ type: Input }]
    };
    return NgSpinnerComponent;
}());
export { NgSpinnerComponent };
if (false) {
    /** @type {?} */
    NgSpinnerComponent.prototype.type;
    /** @type {?} */
    NgSpinnerComponent.prototype.size;
    /** @type {?} */
    NgSpinnerComponent.prototype.color;
}
